### PR TITLE
1853 make getter setter checking stricter

### PR DESCRIPTION
**Pull request by AlexandrKhabarov**
# I have made things!

## Checklist

- [X] I have double checked that there are no unrelated changes in this pull request (old patches, accidental config files, etc)
- [X] I have created at least one test case for the changes I have made
- [X] I have updated the documentation for the changes I have made
- [X] I have added my changes to the `CHANGELOG.md`

### CODE REVIEW

**Comment by AlexandrKhabarov** [comment]
@sobolevn 
Nikita, can you review my request?

File: wemake_python_styleguide/logic/tree/getters_setters.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+from wemake_python_styleguide.compat.aliases import FunctionNodes
+from wemake_python_styleguide.constants import UNUSED_PLACEHOLDER
+from wemake_python_styleguide.logic import nodes
+from wemake_python_styleguide.logic.naming import name_nodes
+from wemake_python_styleguide.logic.tree import classes
+from wemake_python_styleguide.types import AnyFunctionDef
+
+#: Prefixes that usually define getters and setters.
+_GetterSetterPrefixes: Final = ('get_', 'set_')
```
Comment (sobolevn) [inline wemake_python_styleguide/logic/tree/getters_setters.py:14]: This should be a `frozenset()`

File: wemake_python_styleguide/visitors/ast/classes.py [hunk 2]
```diff
@@ -118,28 +119,21 @@ def _is_correct_base_class(self, base_class: ast.AST) -> bool:
...
-                        method,
-                        text=method.name,
-                    ),
-                )
+        getters_and_setters = set(filter(
+            lambda getter_setter: functions.is_method(
+                getattr(getter_setter, 'function_type', None),
+            ),
+            getters_setters.find_paired_getters_and_setters(node),
+        )).union(set(
```
Comment (sobolevn) [inline wemake_python_styleguide/visitors/ast/classes.py:127]: Can you please write two comments for both `.union` parts?

Comment (sobolevn) [reply wemake_python_styleguide/visitors/ast/classes.py:127]: Basically, we need to explain why do we have two `set`s here.

**Review by sobolevn — CHANGES_REQUESTED** [review]
Thanks a lot!

File: wemake_python_styleguide/visitors/ast/classes.py (hunk 2 above)
Comment (AlexandrKhabarov) [reply wemake_python_styleguide/visitors/ast/classes.py]: Done

File: wemake_python_styleguide/logic/tree/getters_setters.py (hunk 1 above)
Comment (AlexandrKhabarov) [reply wemake_python_styleguide/logic/tree/getters_setters.py]: Done

File: tests/test_visitors/test_ast/test_classes/test_getter_setter.py [hunk 3]
```diff
@@ -370,7 +380,7 @@ def test_class_attributes_getter_setter(
 @pytest.mark.parametrize('access', ['', '_', '__'])
 @pytest.mark.parametrize(('first', 'second', 'third'), [
     ('attribute', 'some', 'other'),
-    ('attribute', 'some', 'some'),
+    ('attribute', 'some', 'another'),
```
Comment (sobolevn) [inline tests/test_visitors/test_ast/test_classes/test_getter_setter.py:383]: Why did you change this line? Was it failing?
Can you please add `('attribute', 'some', 'some'),` as well?

File: wemake_python_styleguide/visitors/ast/classes.py [hunk 4]
```diff
@@ -118,28 +119,21 @@ def _is_correct_base_class(self, base_class: ast.AST) -> bool:
...
-        for method in classes.find_getters_and_setters(node):
-            if method.name[classes.GETTER_LENGTH:] in attributes_stripped:
-                self.add_violation(
-                    oop.UnpythonicGetterSetterViolation(
-                        method,
-                        text=method.name,
-                    ),
-                )
+        getters_and_setters = set(filter(
+            lambda getter_setter: functions.is_method(
```
Comment (sobolevn) [inline wemake_python_styleguide/visitors/ast/classes.py:123]: Sorry, I still don't understand this part.

What I see here is basically a `filter` over `getters_setters.find_paired_getters_and_setters(node)` that finds all methods (which I guess is true for 100% of them, am I right?).

Comment (AlexandrKhabarov) [reply wemake_python_styleguide/visitors/ast/classes.py:123]: As I understand from description of task, we need to find paired methods of classes for more strict checking. We can have nested getters and setters and this situation should pass checking.

File: tests/test_visitors/test_ast/test_classes/test_getter_setter.py (hunk 3 above)
Comment (AlexandrKhabarov) [reply tests/test_visitors/test_ast/test_classes/test_getter_setter.py:383]: We use class_mixed template third and firth arguments are used as suffixes for methods. 
If I will return some, some then I will get violation.

File: wemake_python_styleguide/visitors/ast/classes.py (hunk 4 above)
Comment (sobolevn) [reply wemake_python_styleguide/visitors/ast/classes.py:123]: Ok, I see! Thanks a lot!

Then we need a test for this, if it is missing right now.

Comment (sobolevn) [reply wemake_python_styleguide/visitors/ast/classes.py:123]: Oh, I see, we have one: https://github.com/wemake-services/wemake-python-styleguide/blob/b27f20a6703f771279d266afe281d054cf187d2f/tests/test_visitors/test_ast/test_classes/test_getter_setter.py#L107

File: tests/test_visitors/test_ast/test_classes/test_getter_setter.py (hunk 3 above)
Comment (sobolevn) [reply tests/test_visitors/test_ast/test_classes/test_getter_setter.py:383]: Got it! Thanks!

**Review by sobolevn — APPROVED** [review]
Thanks a lot for your hard work 💪 

I am going to merge this into `0.16`

**Comment by AlexandrKhabarov** [comment]
Closes #1853

**Pull request merged**